src/node_url.cc

```diff
@@ -52,51 +52,20 @@ using v8::Value;
     }                                                                         \
   }
 
-#define CANNOT_BE_BASE() url.flags |= URL_FLAGS_CANNOT_BE_BASE;
-#define INVALID_PARSE_STATE() url.flags |= URL_FLAGS_INVALID_PARSE_STATE;
-#define SPECIAL()                                                             \
-  {                                                                           \
-    url.flags |= URL_FLAGS_SPECIAL;                                           \
-    special = true;                                                           \
-  }
-#define TERMINATE()                                                           \
-  {                                                                           \
-    url.flags |= URL_FLAGS_TERMINATED;                                        \
-    goto done;                                                                \
-  }
-#define URL_FAILED()                                                          \
-  {                                                                           \
-    url.flags |= URL_FLAGS_FAILED;                                            \
-    goto done;                                                                \
-  }
-
-#define CHECK_FLAG(flags, name) (flags & URL_FLAGS_##name) /* NOLINT */
-
-#define IS_CANNOT_BE_BASE(flags) CHECK_FLAG(flags, CANNOT_BE_BASE)
-#define IS_FAILED(flags) CHECK_FLAG(flags, FAILED)
-
-#define DOES_HAVE_SCHEME(url) CHECK_FLAG(url.flags, HAS_SCHEME)
-#define DOES_HAVE_USERNAME(url) CHECK_FLAG(url.flags, HAS_USERNAME)
-#define DOES_HAVE_PASSWORD(url) CHECK_FLAG(url.flags, HAS_PASSWORD)
-#define DOES_HAVE_HOST(url) CHECK_FLAG(url.flags, HAS_HOST)
-#define DOES_HAVE_PATH(url) CHECK_FLAG(url.flags, HAS_PATH)
-#define DOES_HAVE_QUERY(url) CHECK_FLAG(url.flags, HAS_QUERY)
-#define DOES_HAVE_FRAGMENT(url) CHECK_FLAG(url.flags, HAS_FRAGMENT)
-
-#define SET_HAVE_SCHEME() url.flags |= URL_FLAGS_HAS_SCHEME;
-#define SET_HAVE_USERNAME() url.flags |= URL_FLAGS_HAS_USERNAME;
-#define SET_HAVE_PASSWORD() url.flags |= URL_FLAGS_HAS_PASSWORD;
-#define SET_HAVE_HOST() url.flags |= URL_FLAGS_HAS_HOST;
-#define SET_HAVE_PATH() url.flags |= URL_FLAGS_HAS_PATH;
-#define SET_HAVE_QUERY() url.flags |= URL_FLAGS_HAS_QUERY;
-#define SET_HAVE_FRAGMENT() url.flags |= URL_FLAGS_HAS_FRAGMENT;
-
 #define UTF8STRING(isolate, str)                                              \
   String::NewFromUtf8(isolate, str.c_str(), v8::NewStringType::kNormal)       \
     .ToLocalChecked()
 
 namespace url {
 
+inline bool CheckFlag(struct url_data* url, enum url_flags flag) {
+  return (url->flags & flag) == flag;
+}
+
+inline void SetFlag(struct url_data* url, enum url_flags flag) {
+  url->flags |= flag;
+}
```

Comment:
hmm... I guess... I prefer the inlined methods personally but I can change it up